watson-developer-cloud/discovery-components
# feat: add PDF viewer with highlighting

#### What do these changes do/fix?
This depends on #237.

This PR is to add `PDFViewerHighlighter` component which is a highlight layer for PDF based on text spans on fields in search result document. This also implements `PDFViewerWithHighlight` which is PDFViewerHighlighter integrated with PDFViewer.

The implementation includes a logic to find calculate bboxes for highlighting. Overview of the logic is in [README.md](https://github.com/watson-developer-cloud/discovery-components/blob/feat/highligh-on-pdf/packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/README.md).

- Add generic utilities used by the logic
   - feat: add types and common utilities 45e5da8
- Add an option to extract HTML source in bbox from HTML field to processDoc
   - feat: add option for bbox text to processDoc c2ab1f6
- Implement the logic to calculate bboxes for highlighting (please refer to [README.md](https://github.com/watson-developer-cloud/discovery-components/blob/feat/highligh-on-pdf/packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/README.md) to get high-level view of the logic)
   - feat: add text layer classes 20e4866
      - Implement `TextLayouts`
   - feat: add highlighting logic and README 839f948
      - Implement `textBoxMapping` and `Highlighter`
- Implement Rect components and storybook
  - feat: add PDF highlight component 67c0ac9




#### How do you test/verify these changes?
- Open the storybook **DocumentPreview > components > PdfViewerWithHighlight > with text selection**,
- In the right pane, select a field except for `html` from the dropdown on the top
- Select a range on the text
- Then, verify that the text on PDF is highlighted

#### Have you documented your changes (if necessary)?

#### Are there any breaking changes included in this pull request?
